flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/source/TestMetadataTableReadableMetrics.java

```diff
@@ -134,8 +130,7 @@ private Table createPrimitiveTable() throws IOException {
             createPrimitiveRecord(
                 false, 2, 2L, Float.NaN, 2.0D, new BigDecimal("2.00"), "2", null, null));
 
-    DataFile dataFile =
-        FileHelpers.writeDataFile(table, Files.localOutput(temp.newFile()), records);
+    DataFile dataFile = FileHelpers.writeDataFile(table, Files.localOutput(temp.toFile()), records);
```

Comment:
this should be doing `File.createTempFile("junit", null, temp.toFile())` instead of `temp.toFile()`, similar to how it was done in #9341. The same needs to be done in L151

Reply:
ack